integration/tests/cook/test_cli.py

```diff
@@ -1783,6 +1818,41 @@ def test_cat_binary_file(self):
         self.assertEqual(0, cp.returncode, cp.stderr)
         self.assertEqual(bytes(i for i in range(0, 256)), cp.stdout)
 
+    def test_cat_with_plugin(self):
+        # User defined plugin to print dummy file content from a file
+        with tempfile.NamedTemporaryFile(suffix='.py', delete=True) as temp:
+            plugin_code = """
+class MockResp:
+    def iter_content(self, chunk_size):
+        yield b"Hello"
+        yield b" "
+        yield b"world!"
+
+def dummy_cat_text(_, __, ___):
+    return MockResp()
```

Comment:
Why is the `MockResp` with `yield`s needed?

Reply:
This is probably where the documentation would help :)

The function `download_file` in mesos.http (which we overwrite) returns a Response object, and we use the `iter_content` function in there to stream the bytes of the file to `cat`. I just wanted to create a quick mock object which had a `iter_content` method to generate "Hello World!"

Reply:
See question above about making this plugin API clearer